packages/gatsby-plugin-offline/src/gatsby-node.js

```diff
@@ -53,7 +59,7 @@ exports.onPostBuild = (args, pluginOptions) => {
         `${process.cwd()}/${rootDir}/offline-plugin-app-shell-fallback/index.html`
       )
     )
-  )
+  ).map(omitPrefix)
 
   const globPatterns = files.concat([
```

Comment:
@davidbailey00 do you think it's a good idea to include 404.html here too? We include it's resources above but not the html file itself

By the way is the 404 page required? Do we know that it will exist?

Reply:
1) We don't need the HTML file itself, for the same reason as described earlier with other HTML pages - the resources for any page can be loaded from the offline shell, without needing to cache the HTML file itself.

   If we're *actually* offline, the offline page will be displayed anyway since we can't guarantee a page hasn't been added since going offline, but caching it allows it to be displayed quickly - even when online, the service worker will load pages via the offline shell.

2) `getResourcesFromHtml` returns an empty array if the file doesn't exist, so nothing will go wrong if the user hasn't created a custom 404 page.

In #7355 I ran through a bunch of manual tests to make sure the offline plugin worked correctly in various circumstances, including the ones above, so I can assure you it works properly even without caching the HTML, or without a custom 404 page. Of course, I forgot to test using path prefixes which is why we have this PR :smile: But apart from that, everything should be working fine.